### PR TITLE
Add input validation in frontend and backend and refactor totalXrp to totalXrpDrops

### LINKED ISSUE #78: Add input validation to bill creation screen (client and server side)

### PR DESCRIPTION
**Pull request by roemba**
Closes #78 

- Fixed the problem that if a `ValidationError` occured, the errors would not be logged anywhere.
- Removed all routes in User where a logged in User could access the full details of another User (these routes were not used anyway) and added `@Authorized()` where it made sense and was missing
- Changed the billling code to use only usernames and not full User objects in autocomplete
- Added input validation to the `BillController`, `UserController` and `TransactionRequestController`
- Refactored `totalXrp` to `totalXrpDrops` and added a utility function so the frontend can easily convert between the two
- Added missing Full Name registration field. Interestingly enough, most of the other code I had apparently already written but I only missed the registration field...
- Made the `Amount` field in the Request page required as well as all the Login fields and all of the Registration fields. You now get a nice warning if you forget to fill in one of the fields.

### CODE REVIEW

**Review by astamagg — COMMENTED** [review]
There is no error warning or response in the ui when I try to create a bill request without the necessary fields.

**Comment by roemba** [comment]
This is because Joost already implemented that the send button does not work if one or more conditions are not met. I simply added the `required` tag to the amount in the bill request form but the other validation logic can only be done in JS.

**Comment by astamagg** [comment]
But do we think that is clear enough for the user?

We can always keep it as is and if we have time add more explicit error handers on the user side

**Comment by roemba** [comment]
That is exactly how I thought we wanted to do this :)

**Review by astamagg — DISMISSED** [review]
Agreed, it looks good, we will add more interaction with the user at a later time

File: src/services/TransactionRequestService.ts [hunk 1]
```diff
@@ -44,7 +44,7 @@ export class TransactionRequestService {
             if (bill.creditor.publickey === addr) {
                 const changes = balanceChanges[addr];
                 for (const change of changes) {
-                    if (change.value === "XRP" && Number(change.value) * 1000000 === tr.totalXrp) {
+                    if (change.value === "XRP" && Number(change.value) * 1000000 === tr.totalXrpDrops) {
```
Comment (jeysbach) [inline src/services/TransactionRequestService.ts]: I think there's a bug here, not related to your changes tho.
I think it should say change.currency === "XRP"

You could also use the util for converting xrp to drops here

Comment (roemba) [reply src/services/TransactionRequestService.ts]: Was indeed a bug, resolved this and changed the code to use XRPUtil

**Review by astamagg — APPROVED** [review]

**Pull request merged**